mods/ra/maps/allies-05a/allies05a.lua

```diff
@@ -237,10 +297,31 @@ InitTriggers = function()
 		local flare = Actor.Create("flare", true, { Owner = greece, Location = ExtractionPath[2] + CVec.New(0, -1) })
 		Trigger.AfterDelay(DateTime.Seconds(7), flare.Destroy)
 		Media.PlaySpeechNotification(greece, "SignalFlare")
-		ExtractionHeli = Reinforcements.ReinforceWithTransport(greece, ExtractionHeliType, nil, ExtractionPath)[1]
 
-		Trigger.OnKilled(ExtractionHeli, function() ussr.MarkCompletedObjective(ussrObj) end)
-		Trigger.OnRemovedFromWorld(Tanya, ExtractTanya)
+		if Map.Difficulty == "Real tough guy" then
+			Reinforcements.ReinforceWithTransport(greece, InsertionHeliType, HeliReinforcements, ExtractionPath, { ExtractionPath[1] })
+			if not Harvester.IsDead then
+				Harvester.FindResources()
+			end
+
+		else
+			ExtractionHeli = Reinforcements.ReinforceWithTransport(greece, ExtractionHeliType, nil, ExtractionPath)[1]
+			local exitPos = CPos.New(ExtractionPath[1].X, ExtractionPath[2].Y)
+
+			Trigger.OnKilled(ExtractionHeli, function() ussr.MarkCompletedObjective(ussrObj) end)
+			Trigger.OnRemovedFromWorld(Tanya, function()
+				ExtractUnits(ExtractionHeli, exitPos, function()
+
+					Media.PlaySpeechNotification("TanyaRescued")
```

Comment:
`Fatal Lua Error: [string "BindingSupport.lua"]:30: Unable to convert parameter 0 to Player`